src/components/TopNav.tsx

```diff
@@ -1,20 +1,322 @@
+                        <ListItemText
+                            primary={
+                                <Link style={Links} to="/posts/project">
+                                    Project
+                                </Link>
+                            }
+                        />
+                    </ListItemButton>
+                </ListItem>
+                <ListItem disablePadding>
+                    <ListItemButton sx={style.textCenter}>
+                        <ListItemText
+                            primary={
+                                <Link style={Links} to="/login">
+                                    SignIn
+                                </Link>
+                            }
+                        />
+                    </ListItemButton>
+                </ListItem>
+                <ListItem disablePadding>
+                    <ListItemButton sx={style.textCenter}>
+                        <ListItemText
+                            primary={
+                                <Link style={Links} to="/register">
+                                    SignUp
+                                </Link>
+                            }
+                        />
+                    </ListItemButton>
+                </ListItem>
+                <ListItem disablePadding>
+                    <ListItemButton sx={style.textCenter}>
+                        <ListItemText
+                            primary={
+                                <Link style={Links} to="/my-account/1">
+                                    Profile
+                                </Link>
+                            }
+                        />
+                    </ListItemButton>
+                </ListItem>
+            </List>
+        </Box>
+    );
+
+    const container =
+        window !== undefined ? () => window().document.body : undefined;
+    
     return (
-        <nav style={{ display: "flex", flexDirection: "column" }}>
-            <Link to="/">Home</Link>
-            <Link to="/posts/learn">Learn</Link>
-            <Link to="/posts/interview">Interview</Link>
-            <Link to="/posts/project">Project</Link>
-            <Link to="/posts/general">General</Link>
-            <Link to="/login">Login</Link>
-            <Link to="/register">Register</Link>
-            <Link to="/my-account">My Account</Link>
-            <Link to="/about-us">About Us</Link>
-        </nav>
+        <>
+            <AppBar position="fixed">
+                <Container maxWidth="xl">
+                    <Toolbar
+                        disableGutters
+                        sx={style.flexSpace.between}
+                    >
+                        <Typography
+                            variant="h6"
+                            noWrap
+                            component="a"
+                            href="/"
+                            sx={{...style.typography, ...style.typography.medium}}
```

Comment:
While this approach works, it's not really the best solution. Typically we should look to React conditional rendering (can be done with theme breakpoints) or even better would be to change the structure to isolate the section which won't always be rendered:
```
<Box>
    <Typography>DevPrep</Typography>
    <Logo />
</Box>
```
then you can use either conditional rendering or the styles object to control the display of the Typography component. Also, the final logo may be a PNG or SVG so keeping it separated also makes it easier to style (or animate) if needed.
This would also reduce code duplication as the existing components are essentially identical except for their content.